src/components/Messages/index.tsx

```diff
@@ -1,26 +1,66 @@
+import { useQuery } from '@apollo/client';
+import Preview from '@components/Messages/Preview';
 import { Card } from '@components/UI/Card';
 import { GridItemEight, GridItemFour, GridLayout } from '@components/UI/GridLayout';
 import MetaTags from '@components/utils/MetaTags';
+import type { Profile } from '@generated/types';
+import { ProfilesDocument } from '@generated/types';
 import isFeatureEnabled from '@lib/isFeatureEnabled';
-import type { Conversation, Stream } from '@xmtp/xmtp-js';
+import type { Conversation, Message } from '@xmtp/xmtp-js';
 import { Client } from '@xmtp/xmtp-js';
-import { useRouter } from 'next/router';
 import type { FC } from 'react';
-import { useEffect, useState } from 'react';
+import { useEffect } from 'react';
 import { APP_NAME } from 'src/constants';
 import Custom404 from 'src/pages/404';
+import Custom500 from 'src/pages/500';
 import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 import { useSigner } from 'wagmi';
 
+export class MessagePreview {
+  profile?: Profile;
+  message?: Message;
+
+  constructor(preview?: MessagePreview) {
+    this.profile = preview?.profile;
+    this.message = preview?.message;
+  }
+}
+
 const Messages: FC = () => {
   const { data: signer } = useSigner();
   const currentProfile = useAppStore((state) => state.currentProfile);
-  const [stream, setStream] = useState<Stream<Conversation>>();
-  const messageState = useMessageStore((state) => state);
-  const { client, setClient, conversations, setConversations, messages, setMessages, setLoading } =
-    messageState;
-  const router = useRouter();
+  const client = useMessageStore((state) => state.client);
+  const setClient = useMessageStore((state) => state.setClient);
+  const isMessagesEnabled = isFeatureEnabled('messages', currentProfile?.id);
+
+  const conversations = useMessageStore((state) => state.conversations);
+  const setConversations = useMessageStore((state) => state.setConversations);
+  const messagePreviews = useMessageStore((state) => state.messagePreviews);
+  const setMessagePreviews = useMessageStore((state) => state.setMessagePreviews);
+
+  const peerAddresses = Array.from(messagePreviews.keys());
+  const { error: profilesError } = useQuery(ProfilesDocument, {
```

Comment:
We should pull this out in a new hook

Reply:
Agree! Was planning to clean this up in the next PR and keep this one strictly to the data mapping for now because there are still moving pieces in the data flow (paginating profiles and adding streaming back). You cool if we clean up the hook/loading/error state in its own PR? I have issues tracking everything and I agree with all your comments 😆

Reply:
Will do